packages/eslint-plugin/src/util/hasOverloadSignatures.ts

```diff
@@ -0,0 +1,90 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import type { RuleContext } from '@typescript-eslint/utils/ts-eslint';
+
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import type { MethodDefinitionWithBody } from './isMethodDefinitionWithBody';
+
+import { getStaticMemberAccessValue } from './misc';
+
+/**
+ * A type unaware variation of `checker.isImplementationOfOverload()`.
+ *
+ * @return true if the function or method node has overload signatures.
+ */
+export function hasOverloadSignatures(
+  node: MethodDefinitionWithBody | TSESTree.FunctionDeclaration,
```

Comment:
As opposed to `FunctionDeclaration`, which has a different node type for its definition (`TSDeclareFunction`), `MethodDefinition` doesn't (`MethodDefinition` seem to either have its `value` be a `FunctionExpression` for implementation or `TSEmptyBodyFunctionExpression` in case of overloading, ignoring type/interface's `TSMethodSignature` which seems like it cannot be part of a class).

I've replaced the previous type of `TSESTree.MethodDefinition` with `MethodDefinitionWithBody` (even though the `.value` key isn't necessary).